### ISSUE TITLE
Why mll included in the pyro_model function in Fully Bayesian GPs Example?

### ISSUE THREAD

**Issue by lk1983823** [cornellius-gp/gpytorch#1140]
In the Fully Bayesian GPs - Sampling Hyperparamters with NUTS example, you design a pyro_model function like this:

```
def pyro_model(x, y):
    model.pyro_sample_from_prior()
    output = model(x)
    loss = mll.pyro_factor(output, y)
    return y

```
I dont know what kind of role “loss = mll.pyro_factor(output, y)” plays in a fully bayesian sampling process? From my learning, I don't think any optimization like MLE or MAP is used in a fully bayesian approach. 
An example called "Regression Model to Predict Divorce Rate" 

> http://pyro.ai/numpyro/bayesian_regression.html

of NumPro tutorials，  also use NUTS to sampling the parameters, while it didn't use any loss function.

So, could anyone tell me? Thanks!

**Comment by gpleiss** [comment]
You are right - the variable name `loss` might be a bit misleading.

Sampling with nuts requires computing the GP's marginal log likelihood (with respect to the sampled parameters) and its derivative. This is what `mll.pyro_factor` returns.

We used the term `loss` to be consistent with our other examples. Derivative-based MCMC samplers (like HMC or NUTS) have updates that are very similar to gradient-descent style updates - so the sampling process isn't too different from the MLE-updates of the parameters in the non-fully-Bayesian examples.

**Comment by lk1983823** [comment]
Thank you

发自我的 iPhone
